src/OpenTK.Compute/OpenCL/CL.cs

```diff
@@ -0,0 +1,1728 @@
+		/// <returns></returns>
+		[DllImport(LibName, EntryPoint = "clReleaseContext")]
+		public static extern CLResultCode ReleaseContext(IntPtr context);
+
+		/// <summary>
+		/// Introduced in OpenCL 1.0
+		/// </summary>
+		/// <param name="context"></param>
+		/// <param name="paramName"></param>
+		/// <param name="paramValueSize"></param>
+		/// <param name="paramValue"></param>
+		/// <param name="paramValueSizeReturned"></param>
+		/// <returns></returns>
+		[DllImport(LibName, EntryPoint = "clGetContextInfo")]
+		public static extern CLResultCode GetContextInfo(IntPtr context, ContextInfo paramName, UIntPtr paramValueSize,
+			byte[] paramValue, out UIntPtr paramValueSizeReturned);
+
+		#endregion
+
+		#region CommandQueue API
+
+		/// <summary>
+		/// Introduced in OpenCL 2.0
+		/// </summary>
+		/// <param name="context"></param>
+		/// <param name="device"></param>
+		/// <param name="properties"></param>
+		/// <param name="errorCode"></param>
+		/// <returns></returns>
+		[DllImport(LibName, EntryPoint = "clCreateCommandQueueWithProperties")]
+		public static extern IntPtr CreateCommandQueueWithProperties(IntPtr context, IntPtr device, IntPtr properties,
+			out CLResultCode errorCode);
+
+		/// <summary>
+		/// Introduced in OpenCL 1.0
+		/// </summary>
+		/// <param name="commandQueue"></param>
+		/// <returns></returns>
+		[DllImport(LibName, EntryPoint = "clRetainCommandQueue")]
+		public static extern CLResultCode RetainCommandQueue(IntPtr commandQueue);
+
+		/// <summary>
+		/// Introduced in OpenCL 1.0
+		/// </summary>
+		/// <param name="commandQueue"></param>
+		/// <returns></returns>
+		[DllImport(LibName, EntryPoint = "clReleaseCommandQueue")]
+		public static extern CLResultCode ReleaseCommandQueue(IntPtr commandQueue);
+
+		/// <summary>
+		/// Introduced in OpenCL 1.0
+		/// </summary>
+		/// <param name="commandQueue"></param>
+		/// <param name="paramName"></param>
+		/// <param name="paramSize"></param>
+		/// <param name="paramValue"></param>
+		/// <param name="paramValueSizeReturned"></param>
+		/// <returns></returns>
+		[DllImport(LibName, EntryPoint = "clGetCommandQueueInfo")]
+		public static extern CLResultCode GetCommandQueueInfo(IntPtr commandQueue, CommandQueueInfo paramName,
+			UIntPtr paramSize, byte[] paramValue, UIntPtr paramValueSizeReturned);
+
+		#endregion
+
+		#region Memory Objects APIs
+
+		/// <summary>
+		/// Introduced in OpenCL 1.0
+		/// </summary>
+		/// <param name="context"></param>
+		/// <param name="flags"></param>
+		/// <param name="size"></param>
+		/// <param name="hostPtr"></param>
+		/// <param name="errorCode"></param>
+		/// <returns></returns>
+		[DllImport(LibName, EntryPoint = "clCreateBuffer")]
+		public static extern IntPtr CreateBuffer(IntPtr context, MemoryFlags flags, UIntPtr size, IntPtr hostPtr,
+			out CLResultCode errorCode);
+
+		public static IntPtr CreateBuffer<T>(IntPtr context, MemoryFlags flags, T[] array, out CLResultCode errorCode)
```

Comment:
I would prefer the usage of Span<T> here(for other arrays the same.

Reply:
At least having an overload that gives a span. Someone might have a use for an array overload

Reply:
True i just recently found out that the overloading is unfortunately a little bit finicky with generics and you'd need to use CreateBuffer<float> to be able to use a Span<T> with a float[] input.
For non generics it doesn't matter though as arrays can be implicitely casted to Span ;)